src/uct/cuda/base/cuda_md.c

```diff
@@ -55,6 +61,64 @@ ucs_status_t uct_cuda_base_get_sys_dev(CUdevice cuda_device,
     return ucs_topo_find_device_by_bus_id(&bus_id, sys_dev_p);
 }
 
+static ucs_status_t
+uct_cuda_base_get_base_addr_alloc_length(uct_cuda_copy_md_t *md,
+                                         CUdevice cuda_device,
+                                         const void *address,
+                                         size_t length,
+                                         void **base_address,
+                                         size_t *alloc_length)
+{
+    static size_t total_bytes[UCT_CUDA_MAX_DEVICES];
+    static char dev_name[UCT_CUDA_MAX_DEVICES][UCT_CUDA_DEV_NAME_MAX_LEN];
+    CUresult cu_err;
+    const char *cu_err_str;
+    CUdeviceptr base_addr;
+    size_t alloc_len;
+    double ratio;
+
+    ucs_spin_lock(&uct_cuda_base_lock);
+
+    if (!strlen(dev_name[cuda_device])) {
+        cu_err = cuDeviceGetName(dev_name[cuda_device],
+                                 sizeof(dev_name[cuda_device]), cuda_device);
+        if (cu_err != CUDA_SUCCESS) {
+            cuGetErrorString(cu_err, &cu_err_str);
+            ucs_error("cuDeviceGetName error: %s", cu_err_str);
+            return UCS_ERR_INVALID_ADDR;
```

Comment:
if can't detect device - warn and fallback to return same base & length as passed

Reply:
This is already done because the calling function sets base_addr and alloc_length to passed address and length by default. If the conditions are right, these values are overwritten.